File: python/rna-transcription/dna.py
def to_rna(nucleotides):
    split_nucleotides = list(nucleotides)
    return "".join(map(convert_nucleotide, split_nucleotides))

def convert_nucleotide(nucleotide):
    if nucleotide == "G":
        return "C"
    elif nucleotide == "C":
        return "G"
    elif nucleotide == "T":
        return "A"
    elif nucleotide == "A":
        return "U"

File: elixir/rna-transcription/dna.exs
defmodule DNA do
  @doc """
  Transcribes a character list representing DNA nucleotides to RNA

  ## Examples

  iex> DNA.to_rna('ACTG')
  'UGAC'
  """
  @spec to_rna([char]) :: [char]
  def to_rna(dna) do
    translate = %{
      ?G => ?C,
      ?C => ?G,
      ?T => ?A,
      ?A => ?U
    }
    Enum.map(dna, fn c -> translate[c] end)
  end
end
